### PR TITLE
feat(worktree): make the row menu reachable on every row, primary included

### PR DESCRIPTION
**Pull request by nhtera**
## Summary

Phase 7 of the worktree feature plan. The row menu was gated to non-primary rows, so a fresh install — where every row is primary — had no reachable row menu at all. Every row now gets the `…` button and right-click; *what* the menu offers is decided by a pure builder from the row's capabilities.

- **`RowCapabilities` + `menu_actions`** — `left_rail/row_menu.rs`. The struct carries the row's OWN project (never the active one); the builder is a pure function with a case table. A primary row gets exactly `Run setup` (when defined), `Open in ▸`, `Copy Path`, `New workspace here` — never `Rename`, `Archive`, `Delete`, or `Merge into`. An archived row gets the two finding rows plus `Unarchive` / `Delete`.
- **One "primary" predicate** — `workspace_ops::is_primary_row` (synthesized `primary:` id OR worktree at the project root), shared by the rail and the menu. Found in review: the two had different definitions, and a real row at the project root would have painted as primary yet been offered `Delete`.
- **`Open in ▸`** — `left_rail/open_in.rs`. Apps from `git.toml [[open_in]]` (`OpenInApp { name, command }`); empty means the built-in list — platform opener plus editors found on the machine — resolved at menu-open time. No second binary resolver: the process PATH is repaired once at boot. Off-macOS the stored command is the path `which` found, quoted, since `Command` will not resolve `code.cmd`.
- **Git pane editor** — add form with a `Presets ▾` dropdown that prefills a known editor, per-app `Remove`, and `Reset` to the built-in list. The first edit materialises the built-in list so a removal sticks. The effective list is cached at open and after edits, not resolved per frame.
- **`Copy Path`** — absolute worktree path, with a toast.
- **`Move to Status ▸`** — `WorkPhase::ALL` plus `Clear`, radio-checked, written through the same store and vocabulary as `oximux worktree set --phase`. No second validator: a `WorkPhase` cannot hold anything the CLI would refuse.
- **`New workspace here`** — the project-group `+`, from the primary row.
- **Merge regression seam** — `merge_ops::merge_target` plus a non-active-project test, matching the guard `Delete` already had.
- Submenus expand in place under their header rather than as a flyout: the rail is too narrow for the pointer to reliably cross a flyout gap.

## Verification

- `RUSTFLAGS=-D warnings cargo clippy --workspace --tests` clean; `cargo test -p oximux-app --lib` 2008 passed; `cargo test -p oximux-settings` 212 passed; `oximux-worktree-ops` and `workspace_create_rollback` green.
- New tests: the menu case table (primary exact list, archived, scripts, merge gate, empty app list), a GPUI paint test drawing the menu for every row kind with each submenu expanded, a Git-pane paint test for both list states, `parse_command` / `add_open_in` / `remove_open_in` edge cases (quoted-empty program refused), `is_primary_row`, `merge_target`, presets parse and cover every detected editor.
- Live-verified in a sandboxed dev instance with a seeded project: primary-row menu, `Open in ▸` expanding to Finder + installed editors (Finder opened at the path), `Copy Path` (clipboard + toast), `New workspace here` (dialog with the project preselected), the full worktree-row menu, `Move to Status ▸` writing `in-progress` with the chip painting on the next tick, the Git pane editor round-tripping to `git.toml`, and the `Presets ▾` prefill.

## Open questions

- Removing every `Open in` app returns to the built-in list — a user cannot have zero apps. Stated in the pane hint.
- `run_workspace_script` opens its terminal tab in the ACTIVE project's panes even for another project's row (pre-existing; now reachable from primary rows).
- Rename resolves the row's own project too but has no test seam like Delete and Merge; its resolution is inline.

## Summary by CodeRabbit

* **New Features**
  * Added workspace row menus with actions for copying paths, changing status, creating workspaces, and opening directories in external apps.
  * Added configurable **Open in** applications in Git settings, including built-in file manager and editor presets.
  * Added support for custom editor commands and platform-aware editor detection.
* **Bug Fixes**
  * Workspace actions, scripts, and merge operations now consistently use the selected workspace’s project context.
  * Improved handling of invalid **Open in** entries by falling back to built-in options.
* **Documentation**
  * Documented workspace menus and **Open in** configuration.

### CODE REVIEW

File: apps/desktop/src/shell/left_rail/open_in.rs [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+use oximux_settings::OpenInApp;
+use oximux_settings::git::GitSettings;
+
+/// The apps the submenu should offer: the configured list, or the built-in
+/// one when nothing is configured.
+pub fn effective_apps(settings: &GitSettings) -> Vec<OpenInApp> {
+    if settings.open_in.is_empty() {
+        default_apps()
+    } else {
+        settings.open_in.clone()
```
[the remark this answers is not in the document]
Comment (nhtera) [reply apps/desktop/src/shell/left_rail/open_in.rs]: Fixed in ac96f940: `effective_apps` now drops an entry with a blank name or a command `parse_command` refuses, and falls back to the built-in list when nothing usable remains. Test: `unusable_configured_entries_are_dropped_and_an_all_unusable_list_falls_back`.

File: apps/desktop/src/shell/left_rail/row_menu.rs [hunk 2]
```diff
@@ -129,36 +199,64 @@ const ARCHIVED_ACTIONS: &[WorkspaceRowAction] =
...
+///   and tag a row that does not appear in the live list.
+/// - A **live** row gets the defined scripts, the finding rows, the status
+///   writer, then the management actions.
+///
+/// `Open in` appears only when there is at least one app to offer.
+fn menu_actions(caps: &RowCapabilities) -> Vec<WorkspaceRowAction> {
+    let open_in = (!caps.open_in.is_empty()).then_some(WorkspaceRowAction::OpenIn);
+    let finding = open_in.into_iter().chain([WorkspaceRowAction::CopyPath]);
+
+    if caps.is_archived {
```
[the remark this answers is not in the document]
Comment (nhtera) [reply apps/desktop/src/shell/left_rail/row_menu.rs:219]: Fixed in ac96f940: an archived row that is also primary keeps `Unarchive` and loses `Delete`. Test: `an_archived_primary_row_keeps_unarchive_but_not_delete`.

File: apps/desktop/src/shell/left_rail/row_menu.rs [hunk 3]
```diff
@@ -129,36 +199,64 @@ const ARCHIVED_ACTIONS: &[WorkspaceRowAction] =
...
+        return finding.chain(ARCHIVED_ACTIONS.iter().copied()).collect();
+    }
+
+    let scripts = SCRIPT_ACTIONS
         .iter()
         .copied()
-        .filter(|a| a.script_kind().is_some_and(|k| avail.has(k)))
+        .filter(|a| a.script_kind().is_some_and(|k| caps.scripts.has(k)))
+        // Only `Run setup` belongs on the project's own checkout.
+        .filter(|a| !caps.is_primary || *a == WorkspaceRowAction::RunSetup);
```
[the remark this answers is not in the document]
Comment (nhtera) [reply apps/desktop/src/shell/left_rail/row_menu.rs:239]: Fixed in ac96f940: `run_workspace_script` now resolves panes by the row's `project_id` (`project_panes_by_project`) instead of `active_project_panes()`, and declines with a log line when that project has no panes in this window. Activating the project first was rejected: it rebuilds the sidebar and switches what is on screen, which running a script should not do.

File: apps/desktop/src/shell/settings_modal/mod.rs [hunk 4]
```diff
@@ -921,6 +954,75 @@ impl SettingsModal {
...
+        name_input.update(cx, |s, cx| s.set_value(app.name.clone(), window, cx));
+        cmd_input.update(cx, |s, cx| s.set_value(app.command.clone(), window, cx));
+        self.git_open_in_notice = None;
+        cx.notify();
+    }
+
+    /// Remove the `idx`-th app of the list the pane is showing. Removing
+    /// from the built-in list materialises it first, so the removal sticks.
+    pub(super) fn remove_open_in_app(&mut self, idx: usize, cx: &mut Context<Self>) {
+        pane_git::remove_open_in(&mut self.git.open_in, idx, open_in::default_apps);
```
[the remark this answers is not in the document]
Comment (nhtera) [reply apps/desktop/src/shell/settings_modal/mod.rs]: Fixed in ac96f940: both `add` and `remove` now materialise `git_open_in_shown` — the list the pane rendered — rather than a fresh `default_apps()`, so the clicked index cannot drift.

File: apps/desktop/src/shell/left_rail/open_in.rs [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+        .map(|(name, bin)| OpenInApp { name: (*name).to_string(), command: (*bin).to_string() })
+        .collect()
+}
+
+#[cfg(not(target_os = "macos"))]
+fn installed_editors() -> Vec<OpenInApp> {
+    CLI_EDITORS
+        .iter()
+        .filter_map(|(name, bin)| {
+            let path = which::which(bin).ok()?;
```
[the remark this answers is not in the document]
Comment (nhtera) [reply apps/desktop/src/shell/left_rail/open_in.rs:128]: Not changed, deliberately. The lookup is four `which` calls, and only on Linux/Windows — macOS checks four bundle paths with `Path::exists`. Each is a handful of `stat`s across `PATH`, on the same order as the settings pane's other per-open reads, and the result is cached for the modal's lifetime. A background task plus a stale-then-refresh state for that would add more code than the pane's whole editor; if a network-mounted `PATH` ever shows up as a real report, the fix is to bound the discovery with a timeout at that point.

File: apps/desktop/src/shell/workspace/workspace_ops.rs [hunk 6]
```diff
@@ -1546,8 +1584,16 @@ impl WorkspaceRoot {
...
         let title = format!("{}: {}", kind.as_str(), workspace.name);
         let script = script.to_string();
-        let Some(panes) = self.active_project_panes() else {
-            tracing::warn!("run_workspace_script: no active project panes");
+        // The ROW's project's panes, not the active project's: the rail shows
+        // every project's rows, and a script for `api`'s worktree must not
+        // land as a tab in `web`'s pane group because `web` happened to be
+        // on screen. A project with no panes in this window is declined.
+        let Some(panes) = self.project_panes_by_project.get(&workspace.project_id).cloned()
+        else {
```
[the remark this answers is not in the document]
Comment (nhtera) [reply apps/desktop/src/shell/workspace/workspace_ops.rs]: Fixed in 9970b8ce: when the row's project has no panes in this window, `run_workspace_script` now activates that project first (`set_active_project` builds the panes) and then opens the terminal there. The no-op path remains only for a project that is not open at all.
